### ISSUE TITLE
Blocking/unblocking old users results in internal server error

### ISSUE THREAD

**Issue by peregrineshahin** [official-stockfish/fishtest#1807]
It seems that we have an issue regarding blocking/unblocking users with a server internal error.
It does not error out for new users but rather it does for old ones.

I suspect it's because of some inconsistent schema variable not existing for old users.
This can be reproduced on DEV also,
Trying to block for example ppigazzini it errors out (unfortunately ;-) ?)
https://tests.stockfishchess.org/user/ctoks

**Comment by vdbergh** [comment]
Yes it must be a field that does not exist for old users. I think we would need the server log to see which one.

**Comment by vdbergh** [comment]
The best solution would be to update the db so that is matches the schema.

**Comment by ppigazzini** [comment]
Most users miss the registration_time key, but few old users miss other keys:
```
bpfliegel missing:
registration_time email tests_repo
iisiraider missing:
registration_time email tests_repo
zirconiumx missing:
registration_time email tests_repo
joster missing:
registration_time email
rstoesser missing:
registration_time
jromang missing:
registration_time
jkiiski missing:
registration_time
rtakker missing:
registration_time
swebber missing:
registration_time tests_repo
nghafoori missing:
registration_time tests_repo
windowstest missing:
registration_time tests_repo
```

**Comment by vdbergh** [comment]
Hmmm. Since you have the script already perhaps you can just extend it to add the missing keys?

**Comment by vdbergh** [comment]
Of course temporarily   we can make all keys optional.

**Comment by ppigazzini** [comment]
Already done on DEV to anonymize the emails. But I would like to get a proper `registration_time` (at the moment I set 10 years ago).
![image](https://github.com/official-stockfish/fishtest/assets/15718418/cdcb3538-5e4d-4d2a-a910-db3afa5e5cac)

**Comment by ppigazzini** [comment]
Here is the script:

```python
from fishtest.rundb import RunDb
from datetime import datetime, timedelta, timezone
rdb = RunDb()
keys = "username", "password", "registration_time", "blocked", "email", "groups", "tests_repo", "machine_limit"
for i, u in enumerate(rdb.userdb.get_users()):
    u['email'] = f"user{i:04d}.user{i:04d}@user{i:04d}"
    if u.get('registration_time') is None:
        u['registration_time'] = datetime.now(timezone.utc) - timedelta(days=3650)
    if u.get('blocked') is None:
        u['blocked'] = False
    if u.get('groups') is None:
        u['groups'] = []
    if u.get('tests_repo') is None:
        u['tests_repo'] = ""
    if u.get('machine_limit') is None:
        u['machine_limit'] = 16

    rdb.userdb.save_user(u)
```

**Comment by vdbergh** [comment]
It's not obvious how to get the correct registration time. I can probably find Gary's email. But most old users will have disappeared.

**Comment by vdbergh** [comment]
A symbolic date sounds good. In principle one could go through the run collection to see when the user first gave CPU time or submitted tests.  Of course this would take a long time.

**Comment by vdbergh** [comment]
You could also use a valid domain such as example.com but perhaps you don't like this.

**Comment by ppigazzini** [comment]
Fixed the email pattern, symbolic date at the start of the 2013.
```bash
git log --reverse
commit f5c16311b0bcf2519b370457fbfa837222eba17c
Author: Gary Linscott <glinscott@gmail.com>
Date:   Sat Feb 9 00:57:20 2013 +0000

    Initial commit
```
![image](https://github.com/official-stockfish/fishtest/assets/15718418/ca85f5b4-985a-4ba7-a418-6c50343ead93)

**Comment by ppigazzini** [comment]
PROD DB should be fixed now.

```python
from datetime import datetime, timezone

from fishtest.rundb import RunDb

rdb = RunDb()
keys = (
    "username",
    "password",
    "registration_time",
    "blocked",
    "email",
    "groups",
    "tests_repo",
    "machine_limit",
)

for u in rdb.userdb.get_users():
    msg = ""
    if u.get("registration_time") is None:
        u["registration_time"] = datetime(2013, 1, 1, 0, 0, 0, tzinfo=timezone.utc)
        msg += " time"
    if u.get("blocked") is None:
        u["blocked"] = False
        msg += " blocked"
    if u.get("email") is None:
        u["email"] = ""
        msg += " email"
    if u.get("groups") is None:
        u["groups"] = []
        msg += " groups"
    if u.get("tests_repo") is None:
        u["tests_repo"] = ""
        msg += " repo"
    if u.get("machine_limit") is None:
        u["machine_limit"] = 16
        msg += " machine"
    if u["registration_time"].tzinfo is None:
        u["registration_time"] = u["registration_time"].replace(tzinfo=timezone.utc)
        msg += " utc"
    if msg:
        rdb.userdb.save_user(u)
        print(f"{u['username']}:{msg}")
```

**Comment by peregrineshahin** [comment]
Closing as the issue is fixed after fixing DB.
